map/routing_manager.cpp

```diff
@@ -271,8 +271,8 @@ void RoutingManager::InsertRoute(routing::Route const & route)
     points.push_back(attrs.GetStart().GetPoint());
     for (auto const & s : segments)
     {
-      points.push_back(s.m_junction.GetPoint());
-      distance += s.m_distFromBeginningMerc;
+      points.push_back(s.GetJunction().GetPoint());
+      distance += s.GetDistFromBeginningMerc();
```

Comment:
Please have a look a the code below
```c++
  for (size_t subrouteIndex = 0; subrouteIndex < route.GetSubrouteCount(); ++subrouteIndex)
  {
    route.GetSubrouteInfo(subrouteIndex, segments);
    Route::SubrouteAttrs attrs;
    route.GetSubrouteAttrs(subrouteIndex, attrs);

    // Fill points.
    double const currentBaseDistance = distance;
    points.clear();
    points.reserve(segments.size() + 1);
    points.push_back(attrs.GetStart().GetPoint());
    for (auto const & s : segments)
    {
      points.push_back(s.GetJunction().GetPoint());
      distance += s.GetDistFromBeginningMerc();
    }
    if (points.size() < 2)
    {
      LOG(LWARNING, ("Invalid subroute. Points number =", points.size()));
      continue;
    }

    auto subroute = make_unique_dp<df::Subroute>();
    subroute->m_polyline = m2::PolylineD(points);
    subroute->m_baseDistance = currentBaseDistance;

```
The value of |distance| is set to |currentBaseDistance| and then  |currentBaseDistance| is used at the line ``` subroute->m_baseDistance = currentBaseDistance;```